packages/core/src/account/smartContractAccount.ts

```diff
@@ -60,13 +60,19 @@ export type SmartContractAccountWithSigner<
   getSigner: () => TSigner;
 };
 
+export const isSmartAccountWithSigner = (
```

Comment:
oh i like this typeguard

Reply:
zod gets really funky with generics and would almost rather we didn't. I think one place where zod would work is parsing like a union type of UserOpRequest or Struct in the EP work actually